OSOE-1107: Reduce dependency on Lombiq.NodeJs.Extensions

sarahelsaig opened the pull request:
[OSOE-1107](https://lombiq.atlassian.net/browse/OSOE-1107)

[OSOE-1107]: https://lombiq.atlassian.net/browse/OSOE-1107?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

sarahelsaig committed:
submodules

sarahelsaig committed:
Update dev submodules

sarahelsaig committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-1107

sarahelsaig committed:
Switch to Npm.Targets.

sarahelsaig committed:
Update Lombiq.Walkthroughs.

sarahelsaig committed:
Remove junk from package.json.

sarahelsaig committed:
Update Lombiq.UIKit.

sarahelsaig committed:
fix readme warnings

sarahelsaig committed:
fix readme warnings

sarahelsaig committed:
Update Lombiq.DataTables.

sarahelsaig committed:
Add missing menu link to UI Kit showcase page.

sarahelsaig committed:
Rewrite textbox-editor.js to not require jQuery.

sarahelsaig committed:
Fix leftover scss in UIKit.

sarahelsaig committed:
Update NPM.Targets NuGet.

sarahelsaig committed:
submodules

sarahelsaig committed:
submodules

sarahelsaig committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-1107

sarahelsaig committed:
Update Lombiq.NodeJs.Extensions NuGet.

sarahelsaig committed:
Update NE nuget.

sarahelsaig committed:
Update NE nuget versions.

sarahelsaig committed:
Update import path.

sarahelsaig committed:
Add project solution linting.

sarahelsaig committed:
Add intentional errors to verify linting.

sarahelsaig committed:
Fix indentation.

sarahelsaig committed:
Use the `>-` expression to break the lines without leaving line breaks in the actual input.

sarahelsaig committed:
Revert

Piedone commented on .github/workflows/build-and-test.yml:
I get this in VS:

![{08FE1EE5-8AE5-403E-B987-C9FBFBE7C1F8}](https://github.com/user-attachments/assets/77288a90-f29a-4771-a219-2eaaad901194)

Even though UITT doesn't use that task.

Despite this, the build succeeds, and when opening the solution again, the error is gone.

This in itself is not nice but can be OK-ish, but might show an actual problem. Isn't this caused by no project references to NPM Targets being added to consumer projects? Only props and targets files. This means that NPM Targets might not be built before its consumer projects (UITT showing here might just be a red herring caused by some VS build idiosyncrasy). I think we had the NE project references for the same reason.

sarahelsaig replied:
Actually UITT [already uses that task](https://github.com/Lombiq/UI-Testing-Toolbox/blob/dev/Lombiq.Tests.UI/Lombiq.Tests.UI.csproj#L109-L110) in dev branch.

Regardless, I've added ProjectReference to Npm.Targets to all projects that import those targets.

Piedone replied:
This still happens, unfortunately, but since everything else works, even the build when this happens, this must be some VS quirk.

Piedone commented on src/Modules/Lombiq.OSOCE.Samples/Navigation.cs:
Thank you! This should rather be added to _Lombiq.UIKit/Showcase_, though.

Piedone commented on .github/workflows/build-and-test.yml:
Also update the NuGetTest solution's package references with the updates you already published, as well as the changed consumer projects so we can see that NuGet publishing correctly embeds the files from the wwwroot folder.

Piedone commented on .github/workflows/build-and-test.yml:
```diff
@@ -60,6 +60,22 @@ jobs:
     name: Codespell
     uses: Lombiq/GitHub-Actions/.github/workflows/codespell.yml@dev
 
+  asset-lint:
+    name: Lint Scripts, Styles, and Texts
+    uses: Lombiq/NodeJs-Extensions/.github/workflows/lint.yml@dev
+    with:
+      scripts: >-
+        src/Modules/Lombiq.DataTables/Lombiq.DataTables,
+        src/Modules/Lombiq.HelpfulExtensions/Lombiq.HelpfulExtensions,
+        src/Modules/Lombiq.UIKit/Lombiq.UIKit,
+        src/Modules/Lombiq.Walkthroughs/Lombiq.Walkthroughs
+      styles-css: >-
+        src/Modules/Lombiq.Privacy/Lombiq.Privacy,
+        src/Modules/Lombiq.HelpfulExtensions/Lombiq.HelpfulExtensions,
+        src/Modules/Lombiq.UIKit/Lombiq.UIKit,
+        src/Modules/Lombiq.JsonEditor/Lombiq.JsonEditor,
+        src/Modules/Lombiq.ChartJs/Lombiq.ChartJs.Samples
```
Also add this to the Windows workflow.

sarahelsaig replied:
Why? The windows workflow don't have codespell either, and this is similar in that it's only static analysis.

Piedone replied:
Linters can behave differently, also due to line ending differences. We've seen this with .NET analyzers and PSScriptAnalyzer too.

sarahelsaig replied:
Have any of those differences ever been anything but false positive? Seems like an unnecessary maintenance burden to me.

Piedone replied:
The .NET ones perhaps mostly yes, but not always, the PSScriptAnalyzer were actual violations.

Since we should be able to run these linters locally too, for a quicker feedback loop, they must work equally well under Windows too, so if there are false positives or any differences between the platforms, we'll need to adapt to them.

Piedone commented on .github/workflows/build-and-test.yml:
```diff
@@ -60,6 +60,22 @@ jobs:
     uses: Lombiq/GitHub-Actions/.github/workflows/codespell.yml@dev
 
+  asset-lint:
+    name: Lint Scripts, Styles, and Texts
```
This would look better with the job being named just Lint, or Lint Assets. Then, since the workflow is named "Lint Scripts, Styles, and Texts", that'll be displayed anyway, but not twice, and we wouldn't have such a long title in the root list:

![{E67D5F68-1DA3-4457-94E1-B320C444B721}](https://github.com/user-attachments/assets/590b117f-0d9a-4db4-8932-10ee159d89d1)


See all the other workflows too.

Piedone committed:
Submodule src/Modules/Lombiq.UIKit updated

Submodule src/Modules/Lombiq.UIKit:
    > 094a7b3 - Typo

[skip ci]

sarahelsaig committed:
Fix readme.

sarahelsaig committed:
Move navigation to UIKit/Showcase.

sarahelsaig committed:
Update nuget solution too.

sarahelsaig committed:
Update DT nuget version.

sarahelsaig committed:
shorten job name

sarahelsaig committed:
UIKit improvements

sarahelsaig committed:
Add UIKit visual verification.

sarahelsaig committed:
CompatibilitySuppressions.xml

sarahelsaig committed:
visual verification image

sarahelsaig committed:
Fix ESLint warnings in NE.

sarahelsaig committed:
submodules

sarahelsaig committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-1107

sarahelsaig committed:
Add linting to windows as well.

sarahelsaig committed:
Necessary to prevent package downgrade.

sarahelsaig committed:
Update rimraf invocation.

sarahelsaig committed:
PackageReference NPM Targets where the props and targets are imported.

sarahelsaig committed:
fix yaml lint

sarahelsaig committed:
submodule

Piedone commented on src/Lombiq.OSOCE.Web/Lombiq.OSOCE.Web.csproj:
```diff
@@ -66,4 +66,11 @@
     <PackageReference Include="OrchardCore.Logging.NLog" Version="$(OrchardCoreVersion)" />
   </ItemGroup>
 
+  <!-- Necessary to prevent package downgrade. -->
```
Where is that coming from, exactly? Can we remove it at one point? Document it here and in NPM Targets.

sarahelsaig replied:
I'm not sure, but after merging from dev it went away, so I've removed it.

sarahelsaig replied:
This (`NU1107: Version conflict detected for Microsoft.Build.Tasks.Core`) showed up again during local builds, but I managed to find and [fix the underlying problem](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/pull/1050/commits/5074232625b0c704bebda48a94f13083caa596a9). It was unrelated to this issue, instead it was caused by [this change](https://github.com/Lombiq/Hosting-Tenants/commit/1facf8b58605188a14762837ddc706ae18aa5107) in the `dev` of Lombiq.Hosting.Tenants. The project was silently importing the NuGet version of NE without anyone realizing until it caused a version conflict with the submodule version of NE.

Piedone replied:
Eh. I guess this is necessary when OCC is mixed into the solution: https://github.com/search?q=org%3ALombiq%20LombiqNodeJsExtensionsPath&type=code Good catch, thank you.

sarahelsaig committed:
update submodule from dev

sarahelsaig committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-1107

sarahelsaig committed:
Fix Microsoft.Build.Tasks.Core version.

sarahelsaig committed:
update submodules to the latest

sarahelsaig committed:
Fix surprise version conflict (NU1107) error.

Terminal capture of the error:
```
$ dotnet build --no-incremental
    /home/sarah/Projects/Open-Source-Orchard-Core-Extensions/src/Modules/Lombiq.Hosting.Tenants/Lombiq.Hosting.Tenants.Maintenance.Tests.UI/Lombiq.Hosting.Tenants.Maintenance.Tests.UI.csproj : error NU1107:
      Version conflict detected for Microsoft.Build.Tasks.Core. Install/reference Microsoft.Build.Tasks.Core 17.12.6 directly to project Lombiq.Hosting.Tenants.Maintenance.Tests.UI to resolve this issue.
       Lombiq.Hosting.Tenants.Maintenance.Tests.UI -> Lombiq.Hosting.Tenants.Maintenance -> Lombiq.NodeJs.Extensions 2.1.3 -> Microsoft.Build.Tasks.Core (>= 17.12.6)
       Lombiq.Hosting.Tenants.Maintenance.Tests.UI -> Lombiq.Tests.UI -> Lombiq.Npm.Targets 1.0.0 -> Microsoft.Build.Tasks.Core (>= 17.13.26).
    /home/sarah/Projects/Open-Source-Orchard-Core-Extensions/src/Lombiq.OSOCE.Web/Lombiq.OSOCE.Web.csproj : error NU1107:
      Version conflict detected for Microsoft.Build.Framework. Install/reference Microsoft.Build.Framework 17.12.6 directly to project Lombiq.OSOCE.Web to resolve this issue.
       Lombiq.OSOCE.Web -> Lombiq.ContentEditors -> Lombiq.NodeJs.Extensions 2.1.3 -> Microsoft.Build.Framework (>= 17.12.6)
       Lombiq.OSOCE.Web -> Lombiq.ChartJs -> Lombiq.Npm.Targets 1.0.0 -> Microsoft.Build.Framework (>= 17.14.7).
    /home/sarah/Projects/Open-Source-Orchard-Core-Extensions/test/Lombiq.UITestingToolbox/Lombiq.Tests.UI.Samples/Lombiq.Tests.UI.Samples.csproj : error NU1107:
      Version conflict detected for Microsoft.Build.Tasks.Core. Install/reference Microsoft.Build.Tasks.Core 17.12.6 directly to project Lombiq.Tests.UI.Samples to resolve this issue.
       Lombiq.Tests.UI.Samples -> Lombiq.OSOCE.Web -> Lombiq.ContentEditors -> Lombiq.NodeJs.Extensions 2.1.3 -> Microsoft.Build.Tasks.Core (>= 17.12.6)
       Lombiq.Tests.UI.Samples -> Lombiq.Tests.UI -> Lombiq.Npm.Targets 1.0.0 -> Microsoft.Build.Tasks.Core (>= 17.13.26).
    /home/sarah/Projects/Open-Source-Orchard-Core-Extensions/test/Lombiq.OSOCE.Tests.UI/Lombiq.OSOCE.Tests.UI.csproj : error NU1107:
      Version conflict detected for Microsoft.Build.Tasks.Core. Install/reference Microsoft.Build.Tasks.Core 17.12.6 directly to project Lombiq.OSOCE.Tests.UI to resolve this issue.
       Lombiq.OSOCE.Tests.UI -> Lombiq.OSOCE.Web -> Lombiq.ContentEditors -> Lombiq.NodeJs.Extensions 2.1.3 -> Microsoft.Build.Tasks.Core (>= 17.12.6)
       Lombiq.OSOCE.Tests.UI -> Lombiq.OrchardCoreApiClient.Tests.UI -> Lombiq.Tests.UI -> Lombiq.Npm.Targets 1.0.0 -> Microsoft.Build.Tasks.Core (>= 17.13.26).
```

sarahelsaig committed:
submodules

sarahelsaig committed:
Update DT.

sarahelsaig committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-1107

sarahelsaig committed:
dummy

sarahelsaig committed:
Enable globbing in rimraf.

sarahelsaig committed:
Upload new Windows image.

Piedone committed:
Adding missing condition to Windows Lint Assets job

[skip ci]

sarahelsaig committed:
update submodules to dev

sarahelsaig committed:
Update Npm.Targets NuGet version to release.

sarahelsaig committed:
Update NE NuGet version to release.
